src/Features/Core/Portable/InitializeParameter/AbstractInitializeMemberFromParameterCodeRefactoringProviderMemberCreation.cs

```diff
@@ -122,18 +125,49 @@ internal abstract partial class AbstractInitializeMemberFromParameterCodeRefacto
             }
         }
 
-        private IFieldSymbol CreateField(
-            IParameterSymbol parameter, ImmutableArray<NamingRule> rules, ImmutableArray<string> parameterNameParts)
+        private async Task<IFieldSymbol> CreateFieldAsync(
+            Document document, SyntaxNode functionDeclaration, IParameterSymbol parameter, ImmutableArray<NamingRule> rules, ImmutableArray<string> parameterNameParts, CancellationToken cancellationToken)
         {
+            var options = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+            var requireAccessibilityModifiers = options.GetOption(CodeStyleOptions.RequireAccessibilityModifiers);
+
             foreach (var rule in rules)
             {
                 if (rule.SymbolSpecification.AppliesTo(SymbolKind.Field, Accessibility.Private))
                 {
                     var uniqueName = GenerateUniqueName(parameter, parameterNameParts, rule);
 
+                    var accessibilityLevel = Accessibility.Private;
+                    if (requireAccessibilityModifiers.Value == AccessibilityModifiersRequired.Never || requireAccessibilityModifiers.Value == AccessibilityModifiersRequired.OmitIfDefault)
+                    {
+                        // C#: always remove modifier for fields if "omit if default" or "never" option is set
+                        if (functionDeclaration.Language == LanguageNames.CSharp)
```

Comment:
might be good to clarify why this is.  specifically, because we're generating a 'private' field, and since 'private' is the default, this will do the right thing.